pkg/controller/cluster/cluster_controller.go

```diff
@@ -622,12 +590,10 @@ func (c *clusterController) syncCluster(key string) error {
 		return err
 	}
 
-	if !reflect.DeepEqual(oldCluster, cluster) {
```

Comment:
Why do we remove `DeepEqual`, will skip updating object if they are `deep equaled`

Reply:
This judgment is meaningless, the time in the condition is always changing, two objects can not be the same

Reply:
Which condition keeps changing?

Reply:
https://github.com/kubesphere/kubesphere/blob/365924e76b9b9fbe00eb8650678678db6dc35f33/pkg/controller/cluster/cluster_controller.go#L610-L618

Reply:
Reconcile will be triggered after the update,  if not compare conditions, it will keep looping, right?

Reply:
Yes, I missed that, but again, the old `DeepEqual` method doesn't really work

Reply:
@wansir Any more comments?

Reply:
Maybe I should revert the status part change, reorganize that part and submit a separate PR.

Reply:
Any update? What do we do with this PR next?

Reply:
> Maybe I should revert the status part change, reorganize that part and submit a separate PR.

I'm currently working on the workspace API, I'll update this soon.